src/libraries/System.Security.Cryptography.Xml/tests/CanonicalXmlEntityReferenceTests.cs

```diff
@@ -0,0 +1,49 @@
+using System.Security.Cryptography;
+using System.Text;
+using System.Xml;
+using Xunit;
+
+namespace System.Security.Cryptography.Xml.Tests
+{
+    public class CanonicalXmlEntityReferenceTests
+    {
+        [Fact]
+        public void Write_WritesExpectedOutput()
+        {
+            var xmlDocument = new XmlDocument();
+            var entityReference = new CanonicalXmlEntityReference("entity", xmlDocument, true);
+            var stringBuilder = new StringBuilder();
+            var anc = new AncestralNamespaceContextManager();
+
+            entityReference.Write(stringBuilder, DocPosition.InsideRootElement, anc);
+
+            Assert.Equal("<!ENTITY entity SYSTEM \"entity\">", stringBuilder.ToString());
+        }
+
+        [Fact]
+        public void WriteHash_WritesExpectedHash()
+        {
+            var xmlDocument = new XmlDocument();
+            var entityReference = new CanonicalXmlEntityReference("entity", xmlDocument, true);
+            var hashAlgorithm = SHA256.Create();
```

Comment:
This is not a place we would use `var`. Please see the [style guide](https://github.com/dotnet/runtime/blob/ffd3b187038a107e29cd425de3a748540a75e33e/docs/coding-guidelines/coding-style.md#L4), particularly point 10. So this should be explicitly typed.

Additionally, since `SHA256` is disposable, we generally try to make sure that we dispose of things, even in unit tests. Please put this in a `using`.